src/UbiHttp.cpp

```diff
@@ -157,79 +166,62 @@ double UbiHTTP::get(const char *device_label, const char *variable_label) {
   }
 
   _client_https_ubi.print(message);
-
-  while (_client_https_ubi.connected()) {
-    if (_client_https_ubi.available()) {
-      String line = _client_https_ubi.readStringUntil('\n');
-      if (line == "\r") {
-        break;
-      }
-    }
-  }
-
+  _client_https_ubi.flush();
+  
   free(message);
   free(path);
-
-  double value = _parseServerAnswer();
-
-  _client_https_ubi.flush();
-  _client_https_ubi.stop();
+  
+  double value = NAN;
+  
+  /* Reads the response from the server */
+  if (waitServerAnswer()) {
+    value = _parseServerAnswer();
+  } else {
+    if (_debug) {
+      Serial.println(F("Could not read server's response"));
+    }
+  }
+  
   return value;
 }
 
 double UbiHTTP::_parseServerAnswer() {
```

Comment:
The comments below apply for the whole function where you are using the String object.